_source/_posts/2022-06-15-terraform-eks-microservices.md

```diff
@@ -0,0 +1,622 @@
+---
+layout: blog_post
+title: "How to Deploy Java Microservices on Amazon EKS Using Terraform and Kubernetes"
+author: deepu-sasidharan
+by: advocate
+communities: [devops, kubernetes, java]
+description: "Deploy a cloud-native Java microservice stack on Amazon EKS using Terraform and Kubernetes."
+tags: [kubernetes, jhipster, spring-boot, spring, java, aws, eks, react, terraform]
+tweets:
+  - "Deploy a #Java microservice stack on Amazon EKS using #Terraform and #Kubernetes"
+  - "Deploy a cloud-native #Java microservice stack to Amazon EKS with #Terraform and #Kubernetes"
+image: blog/jhipster-k8s-eks-terraform/cover-alt.jpg
+github: https://github.com/oktadev/okta-jhipster-k8s-eks-microservices-example
+type: conversion
+---
+
+When it comes to infrastructure, public clouds are the most popular choice these days, especially Amazon Web Services (AWS). If you are in one of those lucky or unlucky (depending on how you see it) teams running microservices, then you need a way to orchestrate their deployments. When it comes to orchestrating microservices, Kubernetes is the de-facto choice. Most public cloud providers also provide managed Kubernetes as a service; for example, Google provides Google Kubernetes Engine (GKE), Microsoft provides Azure Kubernetes Service (AKS), and Amazon provides Amazon Elastic Kubernetes Service (EKS).
+
+This doesn't mean that deploying and managing microservices on the public cloud is easy; they come with their own challenges and pain. This is especially true for Amazon EKS, which in my opinion, is the hardest to use Kubernetes service but also one of the most flexible. This is because EKS is nothing but some clever orchestrations doing a complex dance on top of other AWS services like EC2, EBS, etc.
```

Comment:
```suggestion
This doesn't mean that deploying and managing microservices on the public cloud is easy; each service comes with its own challenges and pain. This is especially true for Amazon EKS, which, in my opinion, is the most difficult Kubernetes service to use, but also one of the most flexible. This is because EKS is nothing but some clever orchestrations doing a complex dance on top of other AWS services like EC2, EBS, etc.
```
easier read, more clarity by avoiding "they" hree

Reply:
updated